modules/core/04-channel/types/packet.go

```diff
@@ -2,22 +2,42 @@ package types
 
 import (
 	"crypto/sha256"
+	"reflect"
 
 	errorsmod "cosmossdk.io/errors"
 
-	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	clienttypes "github.com/cosmos/ibc-go/v9/modules/core/02-client/types"
 	host "github.com/cosmos/ibc-go/v9/modules/core/24-host"
 	"github.com/cosmos/ibc-go/v9/modules/core/exported"
 )
 
-// CommitPacket returns the packet commitment bytes. The commitment consists of:
+// CommitPacket returns the packet commitment bytes based on
+// the ProtocolVersion specified in the Packet. The commitment
+// must commit to all fields in the packet apart from the source port
+// source channel and sequence (which will be committed to in the packet commitment key path)
+// and the ProtocolVersion which is defining how to hash the packet fields.
+// NOTE: The commitment MUSTs be a fixed length preimage to prevent relayers from being able
+// to malleate the packet fields and create a commitment hash that matches the original packet.
+func CommitPacket(packet Packet) []byte {
+	switch packet.ProtocolVersion {
+	case IBC_VERSION_UNSPECIFIED, IBC_VERSION_CLASSIC:
+		return CommitClassicPacket(packet)
+	case IBC_VERSION_EUREKA:
+		return CommitEurekaPacket(packet)
+	default:
+		panic("not implemented")
+	}
+}
+
+// CommitClassicPacket returns the classic packet commitment bytes. The commitment consists of:
 // sha256_hash(timeout_timestamp + timeout_height.RevisionNumber + timeout_height.RevisionHeight + sha256_hash(data))
 // from a given packet. This results in a fixed length preimage.
+// NOTE: A fixed length preimage is ESSENTIAL to prevent relayers from being able
+// to malleate the packet fields and create a commitment hash that matches the original packet.
 // NOTE: sdk.Uint64ToBigEndian sets the uint64 to a slice of length 8.
-func CommitPacket(cdc codec.BinaryCodec, packet Packet) []byte {
```

Comment:
The codec was never used in a version of IBC. Think it was used in early pre-launch implementations but got changed before launch. the function signature never changed tho